diff --git a/dist/main.js b/dist/main.js
index f6ad461a13705c80bc7935cf58d7e3e87900785e..02b75093513bdd1b238ca3b3f132c7c63370e111 100644
--- a/dist/main.js
+++ b/dist/main.js
@@ -66,52 +66,57 @@ function vitePluginArraybuffer() {
       }
       return;
     },
-    load(id) {
-      if (id === "virtual:decode-64") {
-        return decode64Raw;
-      }
-      return;
-    },
-    async transform(_src, id) {
-      if (id.endsWith("?arraybuffer")) {
-        const file = id.slice(0, -12);
-        this.addWatchFile(file);
-        return {
-          code: `export default new Uint8Array([${new Uint8Array(await promises.readFile(file)).join(",")}]).buffer`,
-          map: { mappings: "" }
-        };
-      }
-      if (id.endsWith("?uint8array")) {
-        const file = id.slice(0, -11);
-        this.addWatchFile(file);
-        return {
-          code: `export default new Uint8Array([${new Uint8Array(await promises.readFile(file)).join(",")}])`,
-          map: { mappings: "" }
-        };
-      }
-      if (id.endsWith("?arraybuffer&base64")) {
-        const file = id.slice(0, -19);
-        this.addWatchFile(file);
-        const buffer = await promises.readFile(file);
-        const b64 = buffer.toString("base64");
-        return {
-          code: `import decode64 from 'virtual:decode-64'
-export default decode64("${b64}").buffer`,
-          map: { mappings: "" }
-        };
-      }
-      if (id.endsWith("?uint8array&base64")) {
-        const file = id.slice(0, -18);
-        this.addWatchFile(file);
-        const buffer = await promises.readFile(file);
-        const b64 = buffer.toString("base64");
-        return {
-          code: `import decode64 from 'virtual:decode-64'
-export default decode64("${b64}")`,
-          map: { mappings: "" }
-        };
+    load: {
+      filter: {
+        id: /(\?arraybuffer|\?uint8array|virtual:decode-64)/
+      },
+      async handler(id) {
+        if (id === "virtual:decode-64") {
+          return decode64Raw;
+        }
+        
+        if (id.endsWith("?arraybuffer")) {
+          const file = id.slice(0, -12);
+          this.addWatchFile(file);
+          return {
+            code: `export default new Uint8Array([${new Uint8Array(await promises.readFile(file)).join(",")}]).buffer`,
+            map: { mappings: "" }
+          };
+        }
+
+        if (id.endsWith("?uint8array")) {
+          const file = id.slice(0, -11);
+          this.addWatchFile(file);
+          return {
+            code: `export default new Uint8Array([${new Uint8Array(await promises.readFile(file)).join(",")}])`,
+            map: { mappings: "" }
+          };
+        }
+
+        if (id.endsWith("?arraybuffer&base64")) {
+          const file = id.slice(0, -19);
+          this.addWatchFile(file);
+          const buffer = await promises.readFile(file);
+          const b64 = buffer.toString("base64");
+          return {
+            code: `import decode64 from 'virtual:decode-64'
+  export default decode64("${b64}").buffer`,
+            map: { mappings: "" }
+          };
+        }
+
+        if (id.endsWith("?uint8array&base64")) {
+          const file = id.slice(0, -18);
+          this.addWatchFile(file);
+          const buffer = await promises.readFile(file);
+          const b64 = buffer.toString("base64");
+          return {
+            code: `import decode64 from 'virtual:decode-64'
+  export default decode64("${b64}")`,
+            map: { mappings: "" }
+          };
+        }
       }
-      return;
     }
   };
 }
